examples/pyplots/auto_subplots_adjust.py

```diff
@@ -1,15 +1,41 @@
 """
-====================
-Auto Subplots Adjust
-====================
+===============================================
+Programmatically controlling subplot adjustment
+===============================================
 
-Automatically adjust subplot parameters. This example shows a way to determine
-a subplot parameter from the extent of the ticklabels using a callback on the
-:doc:`draw_event</users/event_handling>`.
+.. note::
 
-Note that a similar result would be achieved using `~.Figure.tight_layout`
-or `~.Figure.set_constrained_layout`; this example shows how one could
-customize the subplot parameter adjustment.
+    This example is primarily intended to show some advanced concepts in
+    Matplotlib.
+
+    If you are only looking for having enough space for your labels, it is
+    almost always simpler and good enough to either set the subplot parameters
+    manually using `.Figure.subplots_adjust`, or use one of the automatic
+    layout mechanisms
+    (:doc:`/tutorials/intermediate/tight_layout_guide`
+    or :doc:`/tutorials/intermediate/constrainedlayout_guide`).
```

Comment:
I think we might as well start swapping these...
```suggestion
    (:doc:`/tutorials/intermediate/constrainedlayout_guide`
    or :doc:`/tutorials/intermediate/tight_layout_guide`).
```

Reply:
sure.